File: tests/codegen/snapshots/typescript/generic_types.ts
type ListLifeGenericResultListLifeItems1 = {
    name: string
    age: number
}

type ListLifeGenericResultListLifeItems2 = {
    name: string
    age: number
}

type ListLifeGenericResultListLife = {
    items1: ListLifeGenericResultListLifeItems1[]
    items2: ListLifeGenericResultListLifeItems2[]
}

type ListLifeGenericResult = {
    list_life: ListLifeGenericResultListLife
}

File: tests/codegen/snapshots/python/generic_types.py
from typing_extensions import TypedDict
from typing import List

class ListLifeGenericResultListLifeItems1(TypedDict):
    name: str
    age: int

class ListLifeGenericResultListLifeItems2(TypedDict):
    name: str
    age: int

class ListLifeGenericResultListLife(TypedDict):
    items1: list[ListLifeGenericResultListLifeItems1]
    items2: list[ListLifeGenericResultListLifeItems2]

class ListLifeGenericResult(TypedDict):
    list_life: ListLifeGenericResultListLife
